daml/Tests/Bond.daml

```diff
@@ -0,0 +1,115 @@
+module Tests.Bond where
+-- ^ This module tests the full lifecycle of a binary option
+
+import Common (Customer, Providers, onboardProviders, onboardCustomer)
+import ContingentClaims.Claim (serialize)
+import DA.Date (date, Month(Jan, Oct))
+import DA.Finance.Types (Id(..), Asset(..))
+import DA.Finance.Asset (AssetDeposit)
+import DA.Finance.Asset.Settlement (AssetSettlementRule(..), AssetSettlement_AddController(..), AssetSettlement_RemoveController(..), AssetSettlement_Transfer(..))
+import Daml.Script
+import DA.Next.Set qualified as Set
+import DA.Map qualified as Map
+import Marketplace.Issuance.Service (RequestOrigination(..), Originate(..), RequestCreateIssuance(..), CreateIssuance(..))
+import Marketplace.Custody.Service (Service, RequestTransferDeposit(..), TransferDeposit(..), RequestLifecycle(..), Lifecycle(..), RequestCreditAccount(..), CreditAccount(..))
+import Marketplace.Issuance.AssetDescription
+import Tests.Utils (bond)
+import DA.Time (time)
+import DA.Assert
+
+usd = "USD"
+
+testBond = script do
+  let date1 = date 2001 Jan 1
+      date2 = date 2002 Oct 31
+      notional = 100.0
+      coupon = 0.1
+      couponPayout = notional * coupon
+      usdAmount = 3000.0
+  (alice, providers, bob, bondCid, bobsDesc, custodyServiceCid, corpActDepositCid) <-
+    requestExerciseBond usdAmount notional coupon [date1, date2]
+
+  -- First lifecycle
+  setTime $ time date1 16 30 00
+  lifecycleReqCid <- submit bob.customer $ exerciseCmd bob.custodyServiceCid RequestLifecycle with
+    assetDepositCid = bondCid
+    choice = bobsDesc.claims
+  bondCid :: _ <- submit providers.bank $ exerciseCmd custodyServiceCid $ Lifecycle with
+    safekeepingDepositCid = corpActDepositCid
+    lifecycleRequestCid = lifecycleReqCid
+    fixings = Map.fromList []
+
+  [(_, bobsOption)] <- queryFilter @AssetDeposit bob.customer (\c -> c.asset.id.label == "BondLabel")
+  1 === bobsOption.asset.id.version
```

Comment:
might be personal preference, but elsewhere in `Tests` these types of asserts are `varToCheck === desiredValue` (which is more readable imo)

Reply:
I was thinking like
```
assertEquals(double expected, double actual)
```
(JUnit)

But reversed.